### DISCUSSION TITLE
Shortcut to switch to different fan configurations?

### DISCUSSION THREAD

**Discussion by KevinK1844** [Rem0o/FanControl.Releases#1191] [Q&A] [answered]
Is there any way to create something like a windows shortcut, batch file, hotkey, etc... that would let me switch between the different saved configurations that I have created? I would like to setup buttons on my Stream Deck that would allow me to change fan configurations more conveniently while in game or whenever I want to switch profiles.

**Comment by Impact123** [comment]
There's a command-line parameter to set the config to be used
> **-c** or **--config** [json config file] command line arg

<https://github.com/Rem0o/FanControl.Releases#installation>
It looks like you have to kill the process first though

**Reply by Rem0o** [reply]
@Impact123  this is just a way to start the app with a pre-set config. To get keyboard shortcuts, would have to use native windows stuff within the app:
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerhotkey

**Reply by Impact123** [reply]
That would be better but their question contained
> something like a [..] batch file

**Comment by Rem0o** [comment]
> It looks like you have to kill the process first though

Won't be necessary in an update coming soon.

**Reply by KevinK1844** [reply]
Awesome to hear, thanks for the reply and I'll look forward to the update.

**Comment by Rem0o** [comment, marked as answer]
See V136. Enjoy.
https://github.com/Rem0o/FanControl.Releases/wiki/Hotkeys---Automate-config-changes-(V136-and-up)

**Reply by KevinK1844** [reply]
This is great, thank you. I've got it working with one minor issue, when triggering the shortcut I'm getting a UAC prompt asking to "...allow this app from an unknown publisher to make changes..." I was able to get around this by making tasks with Windows Task Scheduler to open FanControl with the added argument for the specific configuration file and running it with highest privileges. This is not a big deal for me but in case anyone else see this and is looking for a work around to avoid the UAC prompt. I was then able to call the task from my Stream Deck which allows me to change configurations on the fly which is exactly what I was looking for. 

Here's a link I used, for anyone who wants to create a task to and then create a shortcut to run that task. For Stream Deck users instead of creating a shortcut as described in the link, I used a System Open in the Stream Deck software and typed in "schtasks /run /TN task name" (no quotes w/ your task name) as the App/File to open.

[FIX: Allow the following program from an unknown publisher](https://windowsreport.com/fix-allow-this-app-from-an-unknown-publisher/#4)
